Libraries/Text/Text.js

```diff
@@ -42,12 +42,9 @@ const Text: React.AbstractComponent<
     onResponderTerminationRequest,
     onStartShouldSetResponder,
     pressRetentionOffset,
-    suppressHighlighting,
```

Comment:
The `supressHighlighting` does not work on macOS w/ NSTextView.
This was also causing a re-render of the Text component which lead to race conditions/crashes when tapping href links.

Reply:
After investigating, highlighting does work on desktop w/o NSTextView. This does however cause the aforementioned crash when opening a native link.

We could either remove highlighting from desktop or try and come up with a version that works w/ NSTextView that does not have the same re-rendering behavior that caused the crashes.

Reply:
What does highlighting do on desktop? Is this different than selectable text? I believe on iOS the highlight behavior is that when you click the Text component it highlights it visually, but it highlights the entire component, without arbitrary selectable text (which is what we need).

Reply:
Text Highlighting on iOS is only working w/ Paper (Fabric support is coming).

The entire text component will highlight when tapped. This is a mobile centric UX, similar to what happens when a tableview row is tapped. 

https://user-images.githubusercontent.com/96719/195733831-11fd5204-49ec-4f91-900b-7823cc234227.mp4

On Desktop, this might be implemented as a hover effect instead of the highlight on press.
For these reasons, it makes sense to remove highlighting from RN desktop on MacOS.

@Saadnajmi Let us know if this all makes sense.

Reply:
Did we already have any existing code trying to make 'suppressHighlighting' work on macOS? I'm happy to leave it as an iOS only prop.

Reply:
`suppressHighlighting` actually worked w/o any customization for macOS. This is because it was using the custom text render from iOS implementation instead on NSTextView. However, this highlight should be handled by `hover` state instead of on `press` which is a very mobile centric UX.

Reply:
I see. While it's "technically" a regression, it sounds like it wasn't really something that should have worked on macOS to begin with. I'm happy to just leave it as is. @mischreiber fyi

Reply:
Yep, seems like a reasonable compromise for the time being.